### PR TITLE
Rework `gossipsub-chat` example.

### PR DESCRIPTION
**Pull request by Jerrody**
And I made little refactoring of `Cargo.toml`.

TODO:
- [ ] Change instruction for the tutorial.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -7,7 +7,7 @@ version = "0.44.0"
 authors = ["Parity Technologies <admin@parity.io>"]
 license = "MIT"
 repository = "https://github.com/libp2p/rust-libp2p"
-keywords = ["peer-to-peer", "libp2p", "networking"]
+keywords = ["peer-to-peer", "libp2p", "networking", "p2p"]
```
Comment (mxinden) [inline Cargo.toml]: For the sake of a clean git history, would you mind doing these unrelated changes in separate pull requests?

File: examples/gossipsub-chat.rs [hunk 2]
```diff
@@ -68,12 +139,25 @@ async fn main() -> Result<(), Box<dyn Error>> {
     let local_peer_id = PeerId::from(local_key.public());
     println!("Local peer id: {:?}", local_peer_id);
 
+    let noise_keys = libp2p::noise::Keypair::<libp2p::noise::X25519Spec>::new()
+        .into_authentic(&local_key)
+        .expect("Signing libp2p-noise static DH keypair failed.");
+
     // Set up an encrypted TCP Transport over the Mplex and Yamux protocols
-    let transport = libp2p::development_transport(local_key.clone()).await?;
+    let transport = libp2p::tcp::TokioTcpConfig::new()
```
Comment (mxinden) [inline examples/gossipsub-chat.rs]: How about using the `tokio_development_transport` function instead?

File: examples/gossipsub-chat.rs [hunk 3]
```diff
@@ -45,21 +45,92 @@
 //! ```
 //!
 //! The two nodes should then connect.
-
-use async_std::io;
```
Comment (mxinden) [inline examples/gossipsub-chat.rs:49]: Why the change to tokio?

**Review by mxinden — COMMENTED** [review]
I don't quite understand the goal of this refactoring. What are you trying to showcase?

File: Cargo.toml (hunk 1 above)
Comment (Jerrody) [reply Cargo.toml]: Alright then!

File: examples/gossipsub-chat.rs (hunk 2 above)
Comment (Jerrody) [reply examples/gossipsub-chat.rs]: Hm, maybe it is better than my variant. I just don't understand what's differences between `development_transport` (`tokio_development_transport`) and `TcpConfig` (or `TokioTcpConfig`).

File: examples/gossipsub-chat.rs (hunk 3 above)
Comment (Jerrody) [reply examples/gossipsub-chat.rs:49]: Because this is a standard in Rust's world. Everybody uses `tokio` and this is the fact. `async-std` lost competition with `tokio` and `tokio` turned out to be more viable.
I accept another point of view, so, I want to see your position, pretty good to have many opinions.

**Comment by Jerrody** [comment]
So, I made changes with `Cargo.toml`.
If my PR will be accepted I need to change instructions for the usage in the example's code.
And maybe make more comments on what things do.

File: examples/gossipsub-chat.rs (hunk 2 above)
Comment (mxinden) [reply examples/gossipsub-chat.rs]: The former works with async-io / async-std, the latter with Tokio.

Comment (Jerrody) [reply examples/gossipsub-chat.rs]: > The former works with async-io / async-std, the latter with Tokio.

Alright, let's change to `tokio_development_transport`!

**Comment by Jerrody** [comment]
> I don't quite understand the goal of this refactoring. What are you trying to showcase?

Sorry, I didn't see this message. The main reason is that for the current example we need to pass explicitly in args the address for connecting to the node. In my "refactored" example don't need to specify in the command line the address of the node, the connection is automatic.

**Comment by Jerrody** [comment]
@mxinden So, I wait for your word mate. What should I change else?

**Comment by thomaseizinger** [comment]
This is obsolete with #2996.
